cabal-install/Distribution/Client/ProjectBuilding.hs

```diff
@@ -337,13 +347,13 @@ foldMInstallPlanDepOrder plan0 visit =
 
     go !results (pkg : pkgs) = do
       -- we go in the right order so the results map has entries for all deps
-      let depresults :: ComponentDeps [b]
+      let depresults :: [b]
           depresults =
-            fmap (map (\ipkgid -> let Just result = Map.lookup ipkgid results
-                                   in result))
-                 (depends pkg)
+            map (\ipkgid -> let Just result = Map.lookup ipkgid results
+                              in result)
+                (nodeNeighbors pkg)
       result <- visit pkg depresults
-      let results' = Map.insert (installedPackageId pkg) result results
+      let results' = Map.insert (nodeKey pkg) result results
```

Comment:
I've been feeling a bit dissatisfied with the naming for the graph API and I think these bits here illustrate that quite well. Previously we had `depends` and `installedPackageId` which (reasonably) clearly indicate what role they play here, but with the Graph API we're replacing those with `nodeNeighbors` and `nodeKey` which in this context are just too generic names. I can see that it's nice to look at the graph API in isolation and use graph-specific terminology, but all uses of the graph api are at minimum dependency graphs, even if they're not graphs of packages so I'd like use to think about renaming the graph stuff to less generic names. Also, where we know the concrete type we don't need to use nodeKey, we could stick with `installedPackageId` / `installedUnitId` etc.